src/session/client-certificate.c

```diff
@@ -102,6 +117,40 @@ read_proc_self_cgroup (size_t *out_length)
   return result;
 }
 
+static
+unsigned long long get_proc_pid_start_time (int dirfd)
+{
+  int fd = openat (dirfd, "stat", O_RDONLY);
+  if (fd < 0)
+    err (EX, "Failed to open /proc/pid/stat");
+
+  char buffer[4096];
+  ssize_t size = read (fd, buffer, sizeof buffer);
+  close (fd);
+  if (size < 0)
+    err (EX, "Failed to read /proc/pid/stat");
+
+  /* start time is the token at index 19 after the '(process name)' entry - since only this
+  * field can contain the ')' character, search backwards for this to avoid malicious
+  * processes trying to fool us; See proc_pid_stat(5) */
+  char *p = strrchr (buffer, ')');
+  if (p == NULL)
+    err (EX, "Failed to find process name in /proc/pid/stat");
+  for (int i = 0; i <= 19; i++) /* NB: ')' is the first token */
+    {
+      p = strchr (p, ' ');
+      if (p == NULL)
+        err (EX, "Failed to find start time in /proc/pid/stat");
+      ++p; /* skip over the space */
+    }
+
+  char *endptr;
+  unsigned long long start_time = strtoull (p, &endptr, 10);
+  if (*endptr != ' ')
+    errx (EX, "Failed to parse start time in /proc/pid/stat from %s", p);
+  return start_time;
```

Comment:
I guess this is the kernel so we don't have to worry about bounds-checking the returned value...

I think we can't have other error cases here.  It's worth noting that `strtoull` eats leading whitespace, so if the field were "empty" we'd potentially end up reading whatever the next field is...

Reply:
It actually doesn't  -- this was failing initially, before I added the `++p; /* skip over the space */` above. I.e. it refused to parse " 1234".

Reply:
I don't think we can impose a meaningful bound here. https://man7.org/linux/man-pages/man5/proc_pid_stat.5.html makes no restriction other than monotonicity.